truelearn/models/_knowledge.py

```diff
@@ -1,84 +1,81 @@
-from __future__ import annotations
-from typing import Iterable, Hashable, Any
+from typing import Iterable, Hashable, Any, Optional, Dict, Tuple
+from typing_extensions import Self
 
-from ._abstract_knowledge import AbstractKnowledgeComponent, AbstractKnowledge
+from ._abstract_knowledge import AbstractKnowledgeComponent
 
 
 class KnowledgeComponent(AbstractKnowledgeComponent):
-    """A concrete class that implements AbstractKnowledgeComponent which represents a knowledge component.
-
-    Parameters
-    ----------
-    mean: float
-    variance: float
-    title: str
-    description: str
-    url: str
-
-    Methods
-    -------
-    update(mean, variance)
-        Update the mean and variance of the KnowledgeComponent
-    clone(mean, variance)
-        Clone the KnowledgeComponent with new mean and variance
-    export(output_format)
-        Export the KnowledgeComponent into some format
-
-    Properties
-    ----------
-    title
-    description
-    url
-    mean
-    variance
-
+    """A concrete class that implements AbstractKnowledgeComponent.
+
+    Attributes:
```

Comment:
Unresolved reference for all of these, this is because they arent 'Attributes' but are 'Properties'